graphql_compiler/compiler/emit_sql.py

```diff
@@ -112,3 +125,154 @@ def _get_output_columns(node, context):
+
+
+def _transform_filter_to_sql(filter_block, filter_query_path, context):
+    """Transform a Filter block to its corresponding SQLAlchemy expression.
+
+    Args:
+        filter_block: Filter, the Filter block to transform.
+        filter_query_path: Tuple[str], the query_path the Filter block applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression equivalent to the Filter.predicate expression.
+    """
+    filter_node = sql_context_helpers.get_node_at_path(filter_query_path, context)
+    expression = filter_block.predicate
+    return _expression_to_sql(expression, filter_node, context)
+
+
+def _expression_to_sql(expression, node, context):
+    """Recursively transform a Filter block predicate to its SQLAlchemy expression representation.
+
+    Args:
+        expression: expression, the compiler expression to transform.
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SqlAlchemy expression equivalent to the passed compiler expression.
+    """
+    _expression_transformers = {
+        expressions.LocalField: _transform_local_field_to_expression,
+        expressions.Variable: _transform_variable_to_expression,
+        expressions.Literal: _transform_literal_to_expression,
+        expressions.BinaryComposition: _transform_binary_composition_to_expression,
+    }
+    expression_type = type(expression)
+    if expression_type not in _expression_transformers:
+        raise NotImplementedError(
+            u'Unsupported compiler expression "{}" of type "{}" cannot be converted to SQL '
+            u'expression.'.format(expression, type(expression)))
+    return _expression_transformers[expression_type](expression, node, context)
+
+
+def _transform_binary_composition_to_expression(expression, node, context):
+    """Transform a BinaryComposition compiler expression into a SQLAlchemy expression.
+
+    Recursively calls _expression_to_sql to convert its left and right sub-expressions.
+
+    Args:
+        expression: expression, BinaryComposition compiler expression.
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression.
+    """
+    if expression.operator in constants.UNSUPPORTED_OPERATOR_NAMES:
+        raise NotImplementedError(
+            u'Filter operation "{}" is not supported by the SQL backend.'.format(
+                expression.operator))
+    sql_operator = constants.SUPPORTED_OPERATORS[expression.operator]
+    left = _expression_to_sql(expression.left, node, context)
+    right = _expression_to_sql(expression.right, node, context)
+    if sql_operator.cardinality == constants.Cardinality.UNARY:
+        # ensure the operator is grabbed from the Column object
```

Comment:
Seems like there should be an assertion here? The assumption appears to be that after this block, `left` is an instance of `Column`.